tests/test_errors.py

```diff
@@ -292,6 +292,19 @@ def finalize(self, cfg: Any) -> None:
         ),
         id="structured:setattr,invalid_type_assigned_to_field",
     ),
+    # structured typecheck
+    pytest.param(
+        Expected(
+            create=lambda: None,
+            op=lambda cfg: OmegaConf.structured(User(age="foo")),  # type: ignore
+            exception_type=ValidationError,
+            msg="Value 'foo' could not be converted to Integer",
+            object_type=dict,
+            key="age",
+            low_level=True,
+        ),
+        id="structured:creation-with-incorrect-key-type",
+    ),
```

Comment:
This is the test case motivated by #435.
The `object_type=dict` comes from passing `node=parent` to the `format_and_raise` function, I think.